src/main/java/pl/psnc/dei/ui/components/batches/DatasetImportComponent.java

```diff
@@ -0,0 +1,308 @@
+        HorizontalLayout horizontalLayout = new HorizontalLayout();
+        europeanaDatasetIdTextField = new TextField();
+        europeanaDatasetIdTextField.addClassName("flex-1");
+        europeanaDatasetIdTextField.setLabel("Europeana Dataset ID");
+        Button button = new Button();
+        button.setText("Check dataset");
+        button.getStyle().set("align-self", "flex-end");
+        button.addClickListener(event -> checkDataset());
+        horizontalLayout.add(europeanaDatasetIdTextField, button);
+        return horizontalLayout;
+    }
+
+    private Component prepareRecordLimitInput() {
+        CheckboxWithNumberInput checkbox = new CheckboxWithNumberInput("Limit number of records to retrieve to");
+        limitRecordsToRetrieveCheckbox = checkbox.getCheckbox();
+        limitRecordsToRetrieveNumberField = checkbox.getNumberField();
+        limitRecordsToRetrieveNumberField.setEnabled(doLimitNumberOfRecordsToRetrieve);
+        limitRecordsToRetrieveCheckbox.addValueChangeListener(event -> {
+            doLimitNumberOfRecordsToRetrieve = event.getValue();
+            limitRecordsToRetrieveNumberField.setEnabled(doLimitNumberOfRecordsToRetrieve);
+        });
+        return checkbox;
+    }
+
+    private Component prepareRecordsSplittingInput() {
+        splitRecordsCheckboxWithNumber = new CheckboxWithNumberInput("Split records into several imports with a maximum size of");
+        splitRecordsCheckboxWithNumber.setEnabled(doCreateImport);
+        splitRecordsCheckbox = splitRecordsCheckboxWithNumber.getCheckbox();
+        splitRecordsNumberField = splitRecordsCheckboxWithNumber.getNumberField();
+        splitRecordsNumberField.setEnabled(doSplitRecords);
+        splitRecordsCheckbox.addValueChangeListener(event -> {
+            doSplitRecords = event.getValue();
+            splitRecordsNumberField.setEnabled(doSplitRecords);
+        });
+        return splitRecordsCheckboxWithNumber;
+    }
+
+    private TextField prepareImportNameInputField() {
+        importNameTextField = new TextField();
+        importNameTextField.setLabel("Import name");
+        importNameTextField.getStyle().set("padding-top", "6px");
+        return importNameTextField;
+    }
+
+    private Button prepareUploadRecordsButton() {
+        Button button = new Button("Upload records");
+        button.addClickListener(event -> uploadRecords());
+        return button;
+    }
+
+    private Checkbox prepareCreateImportCheckbox() {
+        createImportCheckbox = new Checkbox();
+        createImportCheckbox.setLabel("Create import");
+        createImportCheckbox.setValue(doCreateImport);
+        createImportCheckbox.addClassName("mt-16px");
+        createImportCheckbox.addValueChangeListener(event -> {
+            doCreateImport = event.getValue();
+            importNameTextField.setEnabled(doCreateImport);
+            splitRecordsCheckboxWithNumber.setEnabled(doCreateImport);
```

Comment:
It will be nice feature if `splitRecordsCheckboxWithNumber` will preserve it state. I mean if `splitRecordsCheckboxWithNumber` is checked, and we uncheck `createImportCheckbox`, and check it once more, then `splitRecordsCheckboxWithNumber` will check by itself.
Same as it is done with value in `splitRecordsCheckboxWithNumber` which preserve its state

Reply:
Sounds OK but I'm not sure I have ever experienced form behaviour like that. I think we could leave it as it is.

Reply:
It is just a little bit unexpected behavior of form, but as I said, it is will be nice feature, so I accept is as it is